src/Core/Injector/Injector.php

```diff
@@ -876,8 +909,10 @@ public function unregisterAllObjects()
      */
     public function get($name, $asSingleton = true, $constructorArgs = null)
     {
+        $object = null;
```

Comment:
I would highly recommend to do that; It'll result in more readable code and less extraneous conditionals through the code, not to mention we can un-do all this nesting.